packages/dds/sequence/src/test/intervalCollection.spec.ts

```diff
@@ -1994,5 +1994,53 @@ describe("SharedString interval collections", () => {
 			assert.equal(interval1.start.slidingPreference, SlidingPreference.BACKWARD);
 			assert.equal(interval1.end.slidingPreference, SlidingPreference.FORWARD);
 		});
+
+		it("slides backward reference to correct position when remove is unacked", () => {
+			sharedString.insertText(0, "ABC");
+
+			// (AB]C
+
+			containerRuntimeFactory.processAllMessages();
+
+			const start = { pos: 0, side: Side.After };
+			const end = { pos: 1, side: Side.After };
+
+			const collection = sharedString.getIntervalCollection("test");
+			collection.add({ end, start });
+
+			assertSequenceIntervals(sharedString, collection, [{ start: 0, end: 1 }]);
+
+			sharedString.removeText(1, 2);
+
+			assertSequenceIntervals(sharedString, collection, [{ start: 0, end: 0 }]);
+
+			containerRuntimeFactory.processAllMessages();
+
+			assertSequenceIntervals(sharedString, collection, [{ start: 0, end: 0 }]);
+		});
+
+		it.skip("slides forward reference to correct position when remove of end of string is unacked", () => {
```

Comment:
Would be fixed by a more general change as mentioned above

Reply:
Something like this would let us share more code between the two codepaths and arguably simplify the helpers: https://github.com/microsoft/FluidFramework/compare/connorskees:fix/interval-sliding...Abe27342:FluidFramework:getSlideToSegmentRefactor

I left your implementation of `referencePositionToLocalPosition` alongside something more similar to what we'd want ideally. Unfortunately, using that code path does cause test problems related to revertibles which I'd need to root cause (which isn't very surprising: it's where we may take dependencies on current behavior of this function).

More generally speaking, I think one thing we'll have to confront here is that:

1. We probably shouldn't do this ad hoc slide for StayOnRemove references (they won't ever be slid unless their endpoint type is changed)
2. We probably should always do this ad hoc slide on read for endpoints of intervals (it's the best approximation we have for where the endpoint will land when there's lots of in-flight ops)
3. The current implementation of intervals accomplishes eventual consistency by changing the slide behavior of its endpoints on ack.

One of the three points above kind of needs to give.

Reply:
talked with Connor offline and I think it's ok to concede that \#1 isn't true above. `localReferencePositionToPosition` has explicit documentation calling this out. In general it's kind of weird for someone to be using StayOnRemove local references in the first place but also attempting to resolve them to positions (rather than deal with segment+offset), so I think this is OK.